core/lib/bfs/bucket/in_mem.rb

```diff
@@ -7,6 +7,36 @@ module Bucket
     class InMem < Abstract
       Entry = Struct.new(:io, :mtime, :content_type, :metadata)
 
+      class Writer < DelegateClass(::StringIO)
+        def initialize(encoding:, &closer)
+          @closer = closer
+
+          sio = StringIO.new
+          sio.set_encoding(encoding)
+          super sio
+        end
+
+        def close
+          close!
+          @closer&.call(self)
+        end
+
+        def close!
+          __getobj__.close
+        end
+
+        def perform(&block)
+          return self unless block
+
+          begin
+            yield self
+            close
```

Comment:
So, basically, on any exception on write (or whatever else is the action done in `.perform` block), you close only StringIO, but not the `@closer` (file being written) so changes are not "committed" (to S3 or other bucket storage), right?